src/EditorFeatures/Next/NavigableSymbols/NavigableSymbolService.Source.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.GoToDefinition;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.CodeAnalysis.Editor.Host;
+
+namespace Microsoft.CodeAnalysis.Editor.NavigableSymbols
+{
+    internal partial class NavigableSymbolService : INavigableSymbolSourceProvider
+    {
+        private partial class Source : INavigableSymbolSource
+        {
+            private readonly ITextView _view;
+            private readonly ITextBuffer _buffer;
+            private readonly IEnumerable<Lazy<IStreamingFindUsagesPresenter>> _presenters;
+
+            public Source(ITextView textView, ITextBuffer textBuffer, IEnumerable<Lazy<IStreamingFindUsagesPresenter>> streamingPresenters)
+            {
+                _view = textView;
+                _buffer = textBuffer;
+                _presenters = streamingPresenters;
+            }
+
+            public void Dispose()
+            {
```

Comment:
store that you're disposed.  Bail immediately when if you are disposed someone called GetNavigableSymbolAsync.